library/EasyPipeline/test/FromFrontend.hs

```diff
@@ -0,0 +1,48 @@
+module FromFrontend where
+
+import qualified Easy
+import qualified Juvix.Core.IR.Types as IR
+import Juvix.Library
+import qualified Juvix.ToCore.Types as Types
+import qualified Test.Tasty as T
+import qualified Test.Tasty.HUnit as T
+
+--------------------------------------------------------------------------------
+-- Top
+--------------------------------------------------------------------------------
+
+top :: T.TestTree
+top =
+  T.testGroup
+    "From Frontend Tests:"
+    [patternVarTests]
+
+--------------------------------------------------------------------------------
+-- tests
+--------------------------------------------------------------------------------
+
+patternVarTests =
+  T.testGroup
+    "pattern Var Tests"
+    [ T.testCase
+        "pattern vars should be tracked"
+        ( do
+            pattern' <- rawPattern
+            pattern' T.@=? rawPatternShouldBe
+        )
+    ]
+  where
+    rawPattern = do
+      Right x <- Easy.coreify "sig foo : int let foo x = x" Easy.defMichelson
```

Comment:
It'd be nice to test `foo x y` for example as it tests the counter as well?

What about `... let foo Nothing = 0; foo (Just x) = x`?

Reply:
I had issues `coreifying` the type declaration, but I did add the other